Added the ability to publish a saved comment

Larspolo opened the pull request:
![image](https://user-images.githubusercontent.com/11822478/140174370-9149a7fb-20c8-4a29-8e7a-c2ed0ff689b7.png)

mjvkeulen reviewed (changes requested):
1. Create a comment using Send & Publish After Grade
2. Edit comment
3. Options are now Send, and Send & Publish

This is confusing, which options should I use to just edit my comment without sending it directly (as was intended).

The use of `Send` in this scenario is just confusing. When editing a nonpublished comment I would expect to see `Save`.

I propose to rename `Send & Publish After Grade` to `Save & Publish After Grade` so we can use `Save` when editing a non-published comment.

Larspolo commented:
The comment is valid, and with it I discovered some [more stuff](https://ejournal.atlassian.net/browse/EJ-1163). Changed the name, also renames "Send & Publish" to just "Publish". Also changed the icon to a save icon

![image](https://user-images.githubusercontent.com/11822478/140582939-292f07c3-77f4-49a7-8b9d-39c3acabc42d.png)

mjvkeulen commented on src/vue/src/components/entry/CommentCard.vue:
```diff
@@ -210,7 +233,7 @@ export default {
                 text: this.comment.text,
                 files: this.comment.files.map((c) => c.id),
-                published: this.comment.published,
+                published: this.comment.published || this.publishOnUpdate === 'true',
```
Is this some limitation of the dropdown button? Why can't we work with `true` as a boolean instead of string?

Larspolo replied:
limitation of the dropdown button. It is based on the object key

mjvkeulen commented on src/vue/src/components/entry/CommentCard.vue:
```diff
@@ -114,22 +114,44 @@
+                            class: 'green-button',
+                        },
+                        true: {
+                            text: 'publish',
```
[NIT] Capitalized anyway but..

Larspolo replied:
Automatically done
